code/client/src/pages/Show/Swap.jsx

```diff
@@ -0,0 +1,268 @@
+    setSelectedTokenSwapFrom({ ...token, value: token.symbol, label: <TokenLabel token={token} /> })
+  }
+
+  const onSelectTokenSwapTo = (token) => {
+    setSelectedTokenSwapTo({ ...token, value: token.symbol, label: <TokenLabel token={token} /> })
+  }
+
+  const onSwapAmountChange = (value) => {
+    const amount = trim(value)
+    if (isNaN(parseInt(amount, 10))) {
+      setSwapAmount(undefined)
+    } else {
+      setSwapAmount(amount)
+    }
+  }
+
+  const confirmSwap = useCallback(() => {
+    const swapAmountBalance = util.toBalance(swapAmount)
+    const computedTokenBalance = getSelectedTokenComputedBalance(selectedTokenSwapFrom, tokenBalances, balance)
+    const tokenBalance = util.toBalance(computedTokenBalance.formatted)
+    const bnTokenBn = tokenBalance.balance
+    const swapAmountBn = swapAmountBalance.balance
+
+    if (swapAmountBalance && bnTokenBn.gte(swapAmountBn)) {
+      // TODO: actual swapping functionalities.
+      console.log(`swapping [${swapAmount}] [${swapAmountBalance.balance}] from [${selectedTokenSwapFrom.name}] to [${selectedTokenSwapTo.name}]`)
+    } else {
+      message.error('Not enough balance to swap')
+    }
+  }, [selectedTokenSwapFrom, selectedTokenSwapTo, tokenBalances, balance, swapAmount])
+
+  const setMaxSwapAmount = () => {
+    const tokenBalance = getSelectedTokenComputedBalance(selectedTokenSwapFrom, tokenBalances, balance)
+
+    const amount = tokenBalance ? tokenBalance.formatted : '0'
+
+    setSwapAmount(amount)
+  }
+
+  const swapSelected = selectedTokenSwapFrom.value !== '' && selectedTokenSwapTo.value !== '' && swapAmount > 0
+
+  return (
+    <>
+      <TallRow align='middle'>
+        <Col span={12}>
+          <Text style={textStyle} type='secondary'>Swap From</Text>
+          <Select
+            showSearch
+            labelInValue
+            style={{ width: 280 }}
+            value={selectedTokenSwapFrom}
+            onSearch={handleSearchCurrentTrackedTokens}
+          >
+            {swapOptions(currentTrackedTokens, onSelectTokenSwapFrom)}
+          </Select>
+        </Col>
+        <Col span={12}>
+          <Text style={textStyle} type='secondary'>Amount (Balance: {selectedTokenBalance})</Text>
+          <InputBox style={amountInputStyle} placeholder='0.00' value={swapAmount} onChange={({ target: { value } }) => onSwapAmountChange(value)} />
+          <Button style={maxButtonStyle} shape='round' onClick={setMaxSwapAmount}>Max</Button>
+        </Col>
+      </TallRow>
+      <TallRow align='middle'>
+        <Col span={12}>
+          <Text style={textStyle} type='secondary'>Swap To</Text>
+          <Select
+            showSearch
+            labelInValue
+            style={{ width: 280 }}
+            value={selectedTokenSwapTo}
+            onSearch={handleSearchSupportedTokens}
+          >
+            {swapOptions(supportedTokens, onSelectTokenSwapTo)}
+          </Select>
+        </Col>
+      </TallRow>
```

Comment:
Need another column displaying the target amount

Reply:
Target amount is based on the exchange rate. Do we have an API for that yet?

Reply:
Not yet, just mock it